src/helper/utils.js

```diff
@@ -0,0 +1,37 @@
+/**
+ * Parse Data Api timestamp field to JS native type.
+ * @param {string} timestamp Timestamp string from Data Api
+ * @returns {Date | number}
+ */
+function parseTimestamp(timestamp) {
+  // Short circuit if the timestamp is already a date
+  if(timestamp instanceof Date){
+    return timestamp;
+  }
+  // Postgres format 2001-01-01 00:00:00 or 2001-01-01 00:00:00.123456
+  // Postgres format +292278994-08-16 23:00:00 or -292275055-05-16 23:00:00 for Infinity values
+  // eslint-disable-next-line no-case-declarations
+  const [, year, month, day, hour, minute, second, millisecond = ''] = timestamp.match(
+    /^([+|-]?\d+)-(\d{1,2})-(\d{1,2}) (\d{1,2}):(\d{1,2}):(\d{1,2})(?:.(\d{1,3}))?(?:\d{1,3})?$/
```

Comment:
```suggestion
    /^([+|-]?\d+)-(\d{1,2})-(\d{1,2}) (\d{1,2}):(\d{1,2}):(\d{1,2})(?:.(\d{1,3}))(?:\d{1,3})$/
```

Reply:
This is copied from the plugin. But I think the ? is needed here since we might not have the last bit in all the date strings.